curator-framework/src/main/java/org/apache/curator/framework/imps/RemoveWatchesBuilderImpl.java

```diff
@@ -0,0 +1,214 @@
+package org.apache.curator.framework.imps;

```

Comment:
I don't think that this is necessarily true is it? We don't want to remove the Watcher reference itself because it may still be watching other paths. If it is no longer referenced then the WeakReference in the NamespaceWatcherMaps internal cache should handle this.